indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -3616,8 +3616,26 @@ private void checkIfStreamInactiveAndTurnSupervisorIdle()
       return;
     }
 
-    Map<PartitionIdType, SequenceOffsetType> latestSequencesFromStream = getLatestSequencesFromStream();
     final long nowTime = DateTimes.nowUtc().getMillis();
+    // if it is the first run and there is no lag observed when compared to the offsets from metadata storage stay idle
+    if (!stateManager.isAtLeastOneSuccessfulRun()) {
+      lastActiveTimeMillis = nowTime;
+      Map<PartitionIdType, SequenceOffsetType> comittedOffsets = getOffsetsFromMetadataStorage();
+      if (comittedOffsets.isEmpty()) {
+        // since there is no previous offsets info available we can skip checking for idleness
+        return;
+      }
+
+      updatePartitionLagFromStream();
+      Map<PartitionIdType, SequenceOffsetType> latestSequencesFromStream = getLatestSequencesFromStream();
+      if (comittedOffsets.equals(latestSequencesFromStream)) {
+        stateManager.maybeSetState(SupervisorStateManager.BasicState.IDLE);
+        previousSequencesFromStream.putAll(comittedOffsets);
+      }
+      return;
```

Comment:
Would it work if we just did this instead?

```suggestion
      // Set previous sequences to the current offsets in metadata store
      previousSequencesFromStream.clear();
      previousSequencesFromStream.putAll(getOffsetsFromMetadataStorage());
      
      // Force update partition lag since the reporting thread might not have run yet
      updatePartitionLagFromStream();
```

The code after this would take care of the rest (once we remove the condition `lastActiveTimeMillis > 0` from the `if` clause).

Reply:
Done.